presto-spi/src/main/java/com/facebook/presto/spi/block/ColumnarArray.java

```diff
@@ -26,6 +26,9 @@ public static ColumnarArray toColumnarArray(Block block)
     {
         requireNonNull(block, "block is null");
 
+        if (block instanceof LazyBlock) {
```

Comment:
Remove handling of `LazyBlock` from all three.

Instead, make `CoercionLazyBlockLoader` strip the `LazyBlock` layer when necessary.